faiss/impl/io.cpp

```diff
@@ -55,6 +55,28 @@ size_t VectorIOReader::operator()(void* ptr, size_t size, size_t nitems) {
     return nitems;
 }
 
+/***********************************************************************
+ * IO Buffer
+ ***********************************************************************/
+
+size_t BufIOReader::operator()(void* ptr, size_t size, size_t nitems) {
```

Comment:
@Thejas-bhat Hard to understand what is going on at every step in this operator override, would you add some commentary along every step that could help.